src/libmeasurement_kit/ndt/utils.cpp

```diff
@@ -94,22 +94,30 @@ report::Entry compute_simple_stats(report::Entry &entry, Var<Logger> logger) {
          * and have multiple S2C entries (e.g., pass both `-T download` and
          * `-T download_ext` to `./measurement_kit`).
          */
+        if (entry["test_s2c"].size() <= 0) {
+            throw std::runtime_error("missing entry");
+        }
         test_s2c = entry["test_s2c"][0];
-        simple_stats["download"] = compute_speed(test_s2c["receiver_data"], logger);
+        simple_stats["download"] = compute_speed(test_s2c["receiver_data"],
+                "download speed", logger);
```

Comment:
You should probably pass only "download" here. Otherwise you print above "Cannot computer download speed speed"